src/lib.rs

```diff
@@ -172,6 +172,19 @@ impl<C: ManageConnection + Send> Pool<C> {
             return self.try_get_connection().await;
         }
 
+        // We go through the max size loop here to:
+        //
+        // 1) Get a new connection.
+        // 2) Grab a connection from the pool.
+        //
+        // In case 1, it will wait in the future and either get the
+        // new connection or return error.  In case 2, we check the validify
```

Comment:
```suggestion
        // new connection or return error.  In case 2, we check the validity
```

Reply:
Thank you, Niels, for catching this up.  Please review the updated version.